setup.py

```diff
@@ -66,53 +85,83 @@ def run_tests(self):
         args = self.tox_args
         if args:
             args = shlex.split(self.tox_args)
+
+        LibDDWaf_Download.download_dynamic_library()
         errno = tox.cmdline(args=args)
         sys.exit(errno)
 
 
-class CMake(BuildExtCommand):
-    def build_extension(self, ext):
-        import shutil
-        import subprocess
-        import tempfile
-
-        to_build = set()
-        # Detect if any source file sits next to a CMakeLists.txt file
-        for source in ext.sources:
-            source_dir = os.path.dirname(os.path.realpath(source))
-            if os.path.exists(os.path.join(source_dir, "CMakeLists.txt")):
-                to_build.add(source_dir)
+class LibDDWaf_Download(BuildPyCommand):
+    @staticmethod
+    def download_dynamic_library():
+        TRANSLATE_SUFFIX = {"Windows": ".dll", "Darwin": ".dylib", "Linux": ".so"}
+        AVAILABLE_RELEASES = {
+            "Windows": ["win32", "x64"],
+            "Darwin": ["arm64", "x86_64"],
+            "Linux": ["aarch64", "x86_64"],
+        }
+        SUFFIX = TRANSLATE_SUFFIX[CURRENT_OS]
 
-        if not to_build:
-            # Build the extension as usual
-            BuildExtCommand.build_extension(self, ext)
+        if os.path.isdir(LIBDDWAF_DOWNLOAD_DIR) and len(os.listdir(LIBDDWAF_DOWNLOAD_DIR)):
```

Comment:
if we have already downloaded it previously?

are we sure we have the right/valid/up-to-date files if this is set?

or the intention is really to do an explicit `python setup.py clean` if you want to invalidate any "cached" files?

Reply:
Well, it downloads the library just for the current OS, so unless we want to do cross-compilation while reusing these files, it should be safe, but as always, if you have a better approach in mind I'd like to hear it